operator/controllers/servicerun_controller.go

```diff
@@ -77,7 +77,7 @@ func (r *ServiceRunReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	logger.Info("Searching for service")
 
 	service := &v1alpha1.Service{}
-	err = r.Get(ctx, types.NamespacedName{Name: serviceRun.Spec.ServiceName, Namespace: serviceRun.Namespace}, service)
+	err = r.Get(ctx, types.NamespacedName{Name: serviceRun.OwnerReferences[0].Name, Namespace: serviceRun.Namespace}, service)
```

Comment:
we should first check if the ownerReference is set before trying to access an item within it

Reply:
Just add the things you need from the service object (checks,...) to the servicerun(instance) object when creating it. Therefore you wouldn't have to query for the service at all and it would stay consistent even if the service object would be changed by a newer version.